### PR TITLE
Security aes alt

### PR DESCRIPTION
**Pull request by pillip8282**

### CODE REVIEW

File: apps/examples/performance/tls_handshake/Makefile [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+SRCS = $(ASRCS) $(CSRCS) $(MAINSRC)
+OBJS = $(AOBJS) $(COBJS)
+
+ifneq ($(CONFIG_BUILD_KERNEL),y)
+  OBJS += $(MAINOBJ)
+endif
+
+ifeq ($(CONFIG_WINDOWS_NATIVE),y)
+  BIN = ..\..\..\libapps$(LIBEXT)
```
Comment (jeongarmy) [inline apps/examples/performance/tls_handshake/Makefile:47]: Could you change `..\..\..\` to `$(APPDIR)` referring to https://github.com/Samsung/TizenRT/pull/5139/commits/019f5995a28cf321bf5a7a9e17ecebe2cbda3776 ?

File: apps/examples/performance/tls_handshake/README.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+examples/tls_handshake
+^^^^^^^^^^^^^^^^^^^^^
+  usage:
+    ex) tls_handshake
```
Comment (jeongarmy) [inline apps/examples/performance/tls_handshake/README.txt:4]: Please update usages for TLS_HANDSHAKE_USAGE.
```
#define TLS_HANDSHAKE_SERVER_USAGE \
	"\ntls_handshake -s\n"             \
	"example: tls_handshake -s\n"

#define TLS_HANDSHAKE_CLIENT_USAGE    \
	"\ntls_handshake -c target_address\n" \
	"example: tls_handshake -c 127.0.0.1\n"

#define TLS_HANDSHAKE_USAGE        \
	"usage: tls_handshake mode\n" \
	"mode : client, server\n" 
```

File: apps/examples/performance/tls_handshake/tls_handshake_main.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include <pthread.h>
+#include "tls_handshake_usage.h"
+
+extern int tls_handshake_server(void);
+extern int tls_handshake_client(char *ipaddr);
+
+int tls_handshake_main(int argc, char **argv)
+{
+	int ret = 0;
+	if (argc == 2) {
```
Comment (jeongarmy) [inline apps/examples/performance/tls_handshake/tls_handshake_main.c:33]: How about changing them like below for code readability?
```
if (argc == 2 && !strncmp("-s", argv[1], 3))) {
  tls_handshake_server();
  return 0;
} else if (argc == 3 && !strncmp("-c", argv[1], 3))) {
 tls_handshake_client(argv[2]);
 return 0;
}

printf("%s\n", TLS_HANDSHAKE_USAGE);
return -1;
```

File: apps/examples/performance/tls_handshake/tls_handshake_usage.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+#define TLS_HANDSHAKE_SERVER_USAGE \
+	"\ntls_handshake -s\n"             \
+	"example: tls_handshake -s\n"
+
+#define TLS_HANDSHAKE_CLIENT_USAGE    \
+	"\ntls_handshake -c target_address\n" \
+	"example: tls_handshake -c 127.0.0.1\n"
+
+#define TLS_HANDSHAKE_USAGE        \
+	"usage: tls_handshake mode\n" \
```
Comment (jeongarmy) [inline apps/examples/performance/tls_handshake/tls_handshake_usage.h:28]: Please change to `tls_handshake <mode>` to separate command and options.

File: apps/examples/performance/tls_handshake/tls_handshake_usage.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#define TLS_HANDSHAKE_SERVER_USAGE \
+	"\ntls_handshake -s\n"             \
+	"example: tls_handshake -s\n"
+
+#define TLS_HANDSHAKE_CLIENT_USAGE    \
+	"\ntls_handshake -c target_address\n" \
```
Comment (jeongarmy) [inline apps/examples/performance/tls_handshake/tls_handshake_usage.h:24]: Please change to `tls_handshake <target_address>` to separate command and options.

File: external/mbedtls/alt/alt_utils.c [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline external/mbedtls/alt/alt_utils.c:19]: I think this is not needed.

File: framework/src/security/security_utils.c [hunk 7]
```diff
@@ -124,46 +123,46 @@ hal_hmac_type secutils_convert_hmacmode_s2h(security_hmac_mode mode)
...
+	case SECLINK_BAD_KEY_PAIR:
 		return SECURITY_ERROR;
-	case HAL_BAD_CERT:
+	case SECLINK_BAD_CERT:
 		return SECURITY_INVALID_CERT_INDEX;
-	case HAL_BAD_CERTKEY_PAIR:
+	case SECLINK_BAD_CERTKEY_PAIR:
 		return SECURITY_ERROR;
-	case HAL_NOT_ENOUGH_MEMORY:
+	case SECLINK_NOT_ENOUGH_MEMORY:
```
Comment (jeongchanKim) [inline framework/src/security/security_utils.c:149]: For memory allocation, only the reason for failure is not enough memory.
Is there any special reason to distinguish `SECLINK_NOT_ENOUGH_MEMORY` and `SECLINK_ALLOC_FAIL`?

**Review by jeongchanKim — COMMENTED** [review]
It is better to remove the logs about `pkbuild`.

File: apps/examples/security_test/seclink/sl_crypto_test.c [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline apps/examples/security_test/seclink/sl_crypto_test.c:19]: This header is not needed.

File: os/drivers/seclink/seclink_drv_utils.h [hunk 9]
```diff
@@ -1,34 +1,81 @@
...
-		SLDRV_LOG(SLDRV_TAG"%s:%d\n", __FILE__, __LINE__);	\
+#define SLDRV_ENTER
+
+#ifdef CONFIG_SECURITY_LINK_DRV_PROFILE
+
+#define SLDRV_CALL(ret, res, method, param)                                         \
+	do {                                                                            \
+		if (se->ops->method) {                                                      \
+			sldrv_timer_handle hnd;                                                 \
+			(void)sldrv_start_time(&hnd);                                           \
```
Comment (jeongchanKim) [inline os/drivers/seclink/seclink_drv_utils.h:51]: If fail to open the timer, we don't need to call `sldrv_get_time` also.
How about checking the return of `sldrv_start_time` and doing next when success to open the timer?

File: apps/examples/performance/tls_handshake/README.txt (hunk 2 above)
Comment (pillip8282) [reply apps/examples/performance/tls_handshake/README.txt:4]: Thanks for your comment. I'll update it.

File: apps/examples/performance/tls_handshake/tls_handshake_main.c (hunk 3 above)
Comment (pillip8282) [reply apps/examples/performance/tls_handshake/tls_handshake_main.c:33]: thanks for comments. Recommended code is more readable.

File: external/mbedtls/alt/alt_utils.c (hunk 6 above)
Comment (pillip8282) [reply external/mbedtls/alt/alt_utils.c:19]: I think adding tinyara/config.h is more safe. if I were not sure about when it is included in code.

File: framework/src/security/security_utils.c (hunk 7 above)
Comment (pillip8282) [reply framework/src/security/security_utils.c:149]: I think it's duplicated. I'll fix it another PR.

File: apps/examples/security_test/seclink/sl_crypto_test.c (hunk 8 above)
Comment (pillip8282) [reply apps/examples/security_test/seclink/sl_crypto_test.c:19]: think adding tinyara/config.h is more safe. if I were not sure about when it is included in code.

File: os/drivers/seclink/seclink_drv_utils.h (hunk 9 above)
Comment (pillip8282) [reply os/drivers/seclink/seclink_drv_utils.h:51]: It's test code. So I think adding exception handling is not necessary.

File: apps/examples/performance/tls_handshake/tls_handshake_main.c (hunk 3 above)
Comment (jeongarmy) [reply apps/examples/performance/tls_handshake/tls_handshake_main.c]: Thank you :)

**Review by an4967 — APPROVED** [review]
LGTM

**Pull request merged**